x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/multivalue/MvContains.java

```diff
@@ -0,0 +1,796 @@
+        ) Expression superset,
+        @Param(
+            name = "subset",
+            type = {
+                "boolean",
+                "cartesian_point",
+                "cartesian_shape",
+                "date",
+                "date_nanos",
+                "double",
+                "geo_point",
+                "geo_shape",
+                "integer",
+                "ip",
+                "keyword",
+                "long",
+                "text",
+                "unsigned_long",
+                "version" },
+            description = "Multivalue expression."
+        ) Expression subset
+    ) {
+        super(source, superset, subset);
+    }
+
+    private MvContains(StreamInput in) throws IOException {
+        super(in);
+    }
+
+    @Override
+    public String getWriteableName() {
+        return ENTRY.name;
+    }
+
+    @Override
+    protected TypeResolution resolveType() {
+        if (childrenResolved() == false) {
+            return new TypeResolution("Unresolved children");
+        }
+
+        TypeResolution resolution = isRepresentableExceptCounters(left(), sourceText(), FIRST);
+        if (resolution.unresolved()) {
+            return resolution;
+        }
+        if (left().dataType() == DataType.NULL) {
+            return isRepresentableExceptCounters(right(), sourceText(), SECOND);
+        }
+        return isType(right(), t -> t.noText() == left().dataType().noText(), sourceText(), SECOND, left().dataType().noText().typeName());
+    }
+
+    @Override
+    public DataType dataType() {
+        return DataType.BOOLEAN;
+    }
+
+    @Override
+    public Nullability nullable() {
+        return Nullability.FALSE;
+    }
+
+    @Override
+    protected MvContains replaceChildren(Expression newLeft, Expression newRight) {
+        return new MvContains(source(), newLeft, newRight);
+    }
+
+    @Override
+    protected NodeInfo<? extends Expression> info() {
+        return NodeInfo.create(this, MvContains::new, left(), right());
+    }
+
+    @Override
+    public Object fold(FoldContext ctx) {
+        return EvaluatorMapper.super.fold(source(), ctx);
+    }
+
+    @Override
+    public ExpressionEvaluator.Factory toEvaluator(ToEvaluator toEvaluator) {
+        var supersetType = PlannerUtils.toElementType(left().dataType());
+        var subsetType = PlannerUtils.toElementType(right().dataType());
+        if (supersetType != ElementType.NULL && subsetType != ElementType.NULL && supersetType != subsetType) {
+            throw new EsqlIllegalArgumentException(
+                "Incompatible data types for MvContains, superset type({}) value({}) and subset type({}) value({}) don't match.",
+                supersetType,
+                left(),
+                subsetType,
+                right()
+            );
+        }
+        if (supersetType == ElementType.NULL || subsetType == ElementType.NULL) {
+            return new MvContainsNullEvaluator(toEvaluator.apply(right()));
```

Comment:
To simplify logics here, if subset is null, we can just return always true. Maybe with a class similar to:

https://github.com/elastic/elasticsearch/blob/f32c3481c292afab091a022c47b6d92fa83dbc15/x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/EvalOperator.java#L160

Reply:
Splitting out into two evaluators.